packages/react-dom/src/client/ToStringValue.js

```diff
@@ -60,7 +86,8 @@ if (enableTrustedTypesIntegration && typeof trustedTypes !== 'undefined') {
         trustedTypes.isScript(value) ||
         trustedTypes.isScriptURL(value) ||
         /* TrustedURLs are deprecated and will be removed soon: https://github.com/WICG/trusted-types/pull/204 */
-        (trustedTypes.isURL && trustedTypes.isURL(value)))
+        (trustedTypes.isURL && trustedTypes.isURL(value)) ||
+        value.$$typeof === REACT_OPAQUE_OBJECT_TYPE)
```

Comment:
This doesn't seem right. This will be toString:ed by the DOM and throw an error anyway. No need to add a special case, is there? Just let it fail below.